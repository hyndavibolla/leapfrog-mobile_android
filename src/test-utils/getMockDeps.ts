import { ENV } from '_constants';
import { GeneralModel, MissionModel } from '_models';
import { RUNTIME_ENV } from '_models/general';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { IHttpService } from '_services/HttpService';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { getInitialState } from '_state_mgmt/GlobalState';
import {
  getUser,
  getGame,
  getActivity_1,
  getActivity_2,
  getActivity_3,
  getMissionCategory_1,
  getMissionCategory_2,
  getMissionCategory_3,
  getMissionKeyword_1,
  getMissionKeyword_2,
  getMissionKeyword_3,
  getMission_1,
  getRewardConfig_1,
  getSlideObject_1,
  getSlideObject_2,
  getSlideCategoryIdList,
  getStreak_1,
  getQuestion_1,
  getQuestion_2,
  getQuestion_3,
  getAnswer_1,
  getAnswer_2,
  getAnswer_3,
  getNewOnMaxMessage_1,
  getNewOnMaxMessage_2,
  getNewOnMaxMessage_3,
  getSailthruMessage_1,
  getLinkedCards_1,
  getLinkedCards_2,
  getLinkedCards_3,
  getLinkedCards_4,
  getNewOnMaxMessage_4,
  getGiftCard,
  getGiftCardDetail,
  getGiftCardBalance
} from '_test_utils/entities';
import { createUUID } from '_utils/create-uuid';

export const getMockDeps = (initialStateGetter = getInitialState): GeneralModel.Deps => ({
  apiService: {
    isTokenInvalid: false,
    getDeviceIdAsync: async () => createUUID(),
    isTokenSetCurrentlyReady: () => true,
    setTokenSet: jest.fn(),
    authenticate: jest
      .fn()
      .mockResolvedValue({ accessToken: 'accessToken', refreshToken: 'refreshToken', accessTokenExpiryTime: 'accessTokenExpiryTime', deviceId: 'deviceId' }),
    enrollCard: jest.fn().mockResolvedValue(undefined),
    fetchRefId: jest.fn().mockResolvedValue('test-ref-id'),
    fetchUserProfile: jest.fn().mockResolvedValue(getUser()),
    fetchGameState: jest.fn().mockResolvedValue(getGame()),
    fetchRewardConfig: jest.fn().mockResolvedValue(getRewardConfig_1()),
    fetchActivityHistory: jest.fn().mockResolvedValue([getActivity_1(), getActivity_2(), getActivity_3()]),
    fetchGiftCardList: jest.fn().mockResolvedValue([getGiftCard()]),
    fetchGiftCardDetail: jest.fn().mockResolvedValue(getGiftCardDetail()),
    fetchGiftCardBalance: jest.fn().mockResolvedValue(getGiftCardBalance()),
    fetchWalletStats: jest.fn().mockResolvedValue({
      totalLifetimeEarnedPoints: 15550,
      monthlySummaries: {
        '2020-02': 1500,
        '2020-03': 2000,
        '2020-04': 100,
        '2020-05': 950,
        '2020-06': 1000,
        '2020-07': 1500
      }
    }),
    fetchMissionCategoryList: jest.fn().mockResolvedValue([getMissionCategory_1(), getMissionCategory_2(), getMissionCategory_3()]),
    fetchMissionKeywordList: jest.fn().mockResolvedValue({
      [MissionModel.KeywordType.CATEGORY]: [getMissionKeyword_1().value],
      [MissionModel.KeywordType.BRAND]: [getMissionKeyword_2().value, getMissionKeyword_3().value]
    }),
    fetchMissionList: jest.fn(async ({ listType }) => ({ userId: 'buttonId', missions: [getMission_1()], title: 'mission endpoint title', listType })),
    registerDeviceInfo: jest.fn(),
    registerUserActivity: jest.fn(),
    logout: jest.fn(),
    getSYWOnlineToken: jest.fn().mockResolvedValue('test-token'),
    sendValidationEmail: jest.fn().mockResolvedValue(undefined),
    fetchStreakList: jest.fn().mockResolvedValue([getStreak_1()]),
    fetchPromptList: jest.fn().mockResolvedValue({
      promptQuestions: [getQuestion_1(), getQuestion_2(), getQuestion_3()],
      answeredQuestions: [getAnswer_1(), getAnswer_2(), getAnswer_3()]
    }),
    updatePromptList: jest.fn(),
    unregisterDeviceInfo: jest.fn(),
    fetchLinkedCardsList: jest.fn().mockResolvedValue({ linkedCards: [getLinkedCards_1(), getLinkedCards_2(), getLinkedCards_3(), getLinkedCards_4()] }),
    activateLocalOffer: jest.fn(),
    fetchAppleWalletPass: jest.fn().mockResolvedValue('test-pass'),
    updateCardStatus: jest.fn()
  } as any,
  appService: {
    getDeviceIdAsync: async () => createUUID()
  } as any,
  crashlyticsService: {
    initialize: jest.fn(),
    log: jest.fn(),
    recordError: jest.fn(),
    setCrashlyticsCollectionEnabled: jest.fn()
  } as ICrashlyticsService,
  httpService: {
    fetch: jest.fn()
  } as IHttpService,
  performanceTrackingService: {
    initialize: jest.fn(),
    measuredFetch: jest.fn(),
    setPerformanceCollectionEnabled: jest.fn()
  } as IPerformanceTrackingService,
  raiseService: {
    fetchSlideBrand: jest.fn().mockResolvedValue([getSlideObject_1(), getSlideObject_2()]),
    fetchSlideCategoryIdList: jest.fn().mockResolvedValue(getSlideCategoryIdList())
  } as any,
  nativeHelperService: {
    alert: {
      alert: jest.fn(),
      prompt: jest.fn()
    },
    appState: { addEventListener: jest.fn(), currentState: 'active' },
    storage: {
      get: jest.fn().mockResolvedValue(null),
      set: jest.fn().mockResolvedValue(null),
      remove: jest.fn().mockResolvedValue(null)
    },
    clipboard: {
      getString: jest.fn(),
      setString: jest.fn()
    },
    dimensions: {
      getWindowWidth: jest.fn().mockReturnValue(750),
      getWindowHeight: jest.fn().mockReturnValue(1000),
      isSmallDevice: false
    },
    deviceInfo: {
      getDeviceId: createUUID,
      getUniqueId: createUUID,
      getVersion: () => '0.0',
      getBuildNumber: () => '0.1',
      getDeviceModel: () => 'iPhone 11',
      getLocales: { languageTag: 'en-US' },
      getBaseOs: async () => 'iOS',
      getSystemVersion: () => '11.0',
      getDeviceToken: () => createUUID(),
      getDeviceTokenAsync: async () => createUUID(),
      getDeviceOsType: async () => 'iOS',
      getAppVersion: () => '0.0.0.1',
      getDeviceLanguage: () => 'en-US',
      getDeviceOsVersion: () => '11.0',
      getFCMTokenAsync: async () => createUUID(),
      getTimeZone: () => 'testing-timezone',
      getReadableVersion: () => '1.1.1.1',
      onFCMTokenRefresh: jest.fn()
    },
    netInfo: {
      getState: jest.fn().mockResolvedValue({ isConnected: true }),
      subscribeToConnectionChanges: jest.fn(),
      waitForConnection: jest.fn().mockResolvedValue(undefined)
    },
    linking: {
      openURL: jest.fn(() => Promise.resolve()),
      canOpenURL: jest.fn(),
      checkURLScheme: jest.fn().mockResolvedValue(true),
      getInitialURL: jest.fn(),
      addEventListener: jest.fn()
    },
    buttonSdk: {
      configure: jest.fn(),
      setUserId: jest.fn(),
      clearData: jest.fn(),
      purchaseRequest: jest.fn()
    },
    platform: {
      os: 'default',
      select: jest.fn(config => config?.default)
    },
    fireBaseMessaging: {
      getToken: jest.fn().mockReturnValue(createUUID())
    },
    hash: {
      buildSHA256: jest.fn().mockResolvedValue('hash')
    },
    restart: jest.fn(),
    cookieManager: {
      clearAll: jest.fn()
    },
    base64: {
      encode: jest.fn((...args) => `b64 encoded: ${JSON.stringify(args)}`),
      decode: jest.fn((...args) => `b64 decoded: ${JSON.stringify(args)}`)
    },
    getRuntimeEnv: jest.fn().mockResolvedValue(RUNTIME_ENV.DEV_TEST),
    messaging: {
      hasPermission: jest.fn().mockReturnValue(true),
      requestPermission: jest.fn()
    },
    pushNotificationsAuthStatus: {
      NOT_DETERMINED: -1,
      DENIED: 0,
      AUTHORIZED: 1,
      PROVISIONAL: 2
    },
    sailthru: {
      getMessages: jest
        .fn()
        .mockResolvedValue([getNewOnMaxMessage_1(), getNewOnMaxMessage_4(), getNewOnMaxMessage_2(), getNewOnMaxMessage_3(), getSailthruMessage_1()]),
      markMessageAsRead: jest.fn().mockResolvedValue(true),
      getInitialNotification: jest.fn(),
      onNotificationTapped: jest.fn(cb => {
        cb({});
        return () => {};
      }) as any
    },
    reactNativePermission: require('react-native-permissions/mock'),
    geolocation: {
      addListener: jest.fn(),
      getCurrentPosition: jest.fn(),
      removeListeners: jest.fn(),
      requestAuthorization: jest.fn(),
      setConfiguration: jest.fn(),
      startObserving: jest.fn(),
      stopObserving: jest.fn()
    }
  } as any,
  stateSnapshot: {
    set: jest.fn(),
    get: jest.fn().mockReturnValue(initialStateGetter())
  },
  logger: {
    log: jest.fn(),
    info: jest.fn(),
    debug: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
    color: jest.fn(),
    assert: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    format: jest.fn()
  } as any,
  eventTrackerService: {
    trackDataUser: jest.fn(),
    isTrackingEnabled: jest.fn().mockResolvedValue(true),
    rZero: {
      trackView: jest.fn(),
      trackEvent: jest.fn(),
      flush: jest.fn(),
      setUser: jest.fn()
    },
    forterSDK: {
      setDevLogsEnabled: jest.fn(),
      getDeviceUniqueID: jest.fn(cb => cb()),
      init: jest.fn((a, b, c) => c()),
      trackNavigation: jest.fn(),
      trackNavigationWithExtraData: jest.fn(),
      trackAction: jest.fn(),
      trackActionWithJSON: jest.fn(),
      fraudTrackNavigationEvent: jest.fn(),
      fraudTrackAction: jest.fn()
    },
    tealiumSDK: {
      track: jest.fn(),
      trackView: jest.fn()
    },
    appsFlyerSDK: {
      updateServerUninstallToken: jest.fn(),
      setCustomerUserId: jest.fn().mockImplementation((_, cb) => cb('success')),
      addDeepLinkEventListener: jest.fn(),
      removeDeepLinkEventListener: jest.fn(),
      clearDeepLinkEventListener: jest.fn(),
      trackConversionEvent: jest.fn()
    }
  } as any,
  ccpaService: {
    showConsentBanner: jest.fn().mockResolvedValue(null),
    showPreferenceCenter: jest.fn()
  } as any,
  remoteConfigService: {
    getValue: jest.fn(async key => ENV.REMOTE_CONFIG.DEFAULT_CONFIG[key]),
    fetchAndActivate: jest.fn().mockResolvedValue(null)
  } as any,
  authService: {
    auth: {
      refreshToken: jest.fn().mockResolvedValue(true)
    },
    webAuth: {
      authorize: jest.fn(),
      clearSession: jest.fn()
    }
  } as any
});
