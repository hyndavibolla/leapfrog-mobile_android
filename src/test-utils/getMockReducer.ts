import { IGlobalState } from '../models/general';
import { combinedReducer } from '../state-mgmt/GlobalState';

export const getMockReducer = (state?: IGlobalState) => jest.fn((s, a) => state || combinedReducer(s, a));
