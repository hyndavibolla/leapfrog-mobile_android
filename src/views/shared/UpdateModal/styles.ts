import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '../../../constants';

export const styles = {
  container: css`
    flex: 1;
    align-items: center;
    justify-content: center;
    padding: 48px 40px;
    width: 100%;
  `,
  swyLogo: css`
    width: 65px;
    height: 65px;
    margin-bottom: 32px;
  `,
  title: css`
    text-align: center;
    margin-bottom: 32px;
  `,
  message: css`
    text-align: center;
    margin-bottom: 32px;
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.BIG};
  `,
  btnContainer: css`
    width: 100%;
  `,
  updateBtnInnerContainer: css`
    padding: 16px 32px;
  `,
  updateBtnText: css`
    font-family: ${FONT_FAMILY.HEAVY};
    font-size: ${FONT_SIZE.SMALLER};
  `,
  dismissBtnInnerContainer: css`
    margin-top: 16px;
    background-color: transparent;
  `,
  dismissBtnText: css`
    color: ${COLOR.PRIMARY_BLUE};
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
  `
};
