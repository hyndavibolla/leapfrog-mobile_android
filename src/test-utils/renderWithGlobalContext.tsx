import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
