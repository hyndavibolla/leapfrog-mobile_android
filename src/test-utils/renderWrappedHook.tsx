import React from 'react';
import { renderHook } from '@testing-library/react-hooks';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWrappedHook = (
  hookGetter: () => any,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const wrapper = ({ children }: any) => (
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderHook(() => hookGetter(), { wrapper }), mockReducer: combinedReducers, deps };
};
