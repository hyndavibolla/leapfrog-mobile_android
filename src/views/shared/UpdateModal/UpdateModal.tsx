import React, { memo, useCallback, useContext } from 'react';
import { View } from 'react-native';
import Image from 'react-native-fast-image';

import { Text } from '../../shared/Text';
import { styles } from './styles';
import { useTestingHelper } from '../../../utils/useTestingHelper';
import { Title, TitleType } from '../Title';
import { Button } from '../Button';
import { GlobalContext } from '../../../state-mgmt/GlobalState';
import { ENV } from '../../../constants';

export interface Props {
  isUpdateRequired: boolean;
  onClose?: () => void;
}

export const UpdateModal = ({ isUpdateRequired, onClose }: Props) => {
  const { getTestIdProps } = useTestingHelper('update-modal');
  const { deps } = useContext(GlobalContext);

  const onUpdate = useCallback(() => {
    deps.nativeHelperService.linking.openURL(
      deps.nativeHelperService.platform.select({ android: ENV.STORE_URL.ANDROID, ios: ENV.STORE_URL.IOS, default: ENV.STORE_URL.IOS })
    );
  }, [deps.nativeHelperService]);

  return (
    <View style={styles.container} {...getTestIdProps('container')}>
      <Image style={styles.swyLogo as any} source={require('../../../assets/shared/swyLogo.png')} resizeMode="cover" />
      <Title style={styles.title} type={TitleType.SECTION}>
        {"Hey! There's a new version of \nSYW MAX available."}
      </Title>
      <Text style={styles.message}>Update now and keep earning points with your favorite brands!</Text>
      <View style={styles.btnContainer}>
        <Button innerContainerStyle={styles.updateBtnInnerContainer} textStyle={styles.updateBtnText} onPress={onUpdate} {...getTestIdProps('update-btn')}>
          Update now
        </Button>
      </View>
      {isUpdateRequired ? null : (
        <Button innerContainerStyle={styles.dismissBtnInnerContainer} textStyle={styles.dismissBtnText} onPress={onClose} {...getTestIdProps('close-btn')}>
          I'll do this later
        </Button>
      )}
    </View>
  );
};

export default memo(UpdateModal);
