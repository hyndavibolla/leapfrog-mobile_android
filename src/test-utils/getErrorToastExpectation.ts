import { act } from 'react-test-renderer';

import { ACTION_TYPE } from '../state-mgmt/core/actions';
import { ToastType } from '../views/shared/Toast';
import { Deps } from '../models/general';
import { wait } from '../utils/wait';
import { renderWrappedHook } from './renderWrappedHook';

export const getErrorToastExpectation = async (hookGetter: () => any, deps: Deps, errorArgs: any[] = []) => {
  const { result, mockReducer } = renderWrappedHook(hookGetter, deps);
  await act(async () => {
    await (result.current[0] as (...args: any[]) => Promise<any>)(...errorArgs);
    await wait(0); // waiting for the useEffect dispatching the toast to run
    expect(mockReducer).toBeCalledWith(
      expect.any(Object),
      expect.objectContaining({
        type: ACTION_TYPE.ADD_TOAST,
        payload: expect.objectContaining({ props: expect.objectContaining({ type: ToastType.ERROR }) })
      })
    );
  });
};
