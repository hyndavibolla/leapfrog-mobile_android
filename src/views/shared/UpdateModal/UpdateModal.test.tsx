import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import { renderWithGlobalContext } from '../../../test-utils/renderWithGlobalContext';

import UpdateModal from './UpdateModal';

describe('UpdateModal', () => {
  it('should render a with a required update', () => {
    const { toJSON } = renderWithGlobalContext(<UpdateModal isUpdateRequired={true} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a without a required update', () => {
    const { toJSON } = renderWithGlobalContext(<UpdateModal isUpdateRequired={false} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have a pressable skip button', () => {
    const fn = jest.fn();
    const { getByTestId } = renderWithGlobalContext(<UpdateModal isUpdateRequired={false} onClose={fn} />);
    fireEvent.press(getByTestId('update-modal-close-btn'));
    expect(fn).toBeCalled();
  });

  it('should have a button to redirect to the stores', () => {
    const { getByTestId, deps } = renderWithGlobalContext(<UpdateModal isUpdateRequired={false} />);
    fireEvent.press(getByTestId('update-modal-update-btn'));
    expect(deps.nativeHelperService.linking.openURL).toBeCalledWith(expect.any(String));
  });
});
