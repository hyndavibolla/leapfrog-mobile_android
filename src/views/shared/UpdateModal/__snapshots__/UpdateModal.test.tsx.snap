// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateModal should render a with a required update 1`] = `
<View
  accessibilityLabel="update-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 48,
      "paddingLeft": 40,
      "paddingRight": 40,
      "paddingTop": 48,
      "width": "100%",
    }
  }
  testID="update-modal-container"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "height": 65,
          "marginBottom": 32,
          "width": 65,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../src/assets/shared/swyLogo.png",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "marginBottom": 32,
          "textAlign": "center",
        },
      ]
    }
    type="sectionHeader"
  >
    Hey! There's a new version of 
SYW MAX available.
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 16,
          "lineHeight": 22,
        },
        Object {
          "color": "#909BA0",
          "fontSize": 16,
          "lineHeight": 24,
          "marginBottom": 32,
          "textAlign": "center",
        },
      ]
    }
  >
    Update now and keep earning points with your favorite brands!
  </Text>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="update-modal-update-btn"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="update-modal-update-btn"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#0066CC",
              "borderBottomLeftRadius": 30,
              "borderBottomRightRadius": 30,
              "borderTopLeftRadius": 30,
              "borderTopRightRadius": 30,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            },
            Object {
              "paddingBottom": 16,
              "paddingLeft": 32,
              "paddingRight": 32,
              "paddingTop": 16,
            },
          ]
        }
      >
        <View
          style={false}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "SFProDisplay-Heavy",
                    "fontSize": 16,
                  },
                  undefined,
                ],
              ]
            }
          >
            Update now
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`UpdateModal should render a without a required update 1`] = `
<View
  accessibilityLabel="update-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 48,
      "paddingLeft": 40,
      "paddingRight": 40,
      "paddingTop": 48,
      "width": "100%",
    }
  }
  testID="update-modal-container"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "height": 65,
          "marginBottom": 32,
          "width": 65,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../src/assets/shared/swyLogo.png",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "marginBottom": 32,
          "textAlign": "center",
        },
      ]
    }
    type="sectionHeader"
  >
    Hey! There's a new version of 
SYW MAX available.
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 16,
          "lineHeight": 22,
        },
        Object {
          "color": "#909BA0",
          "fontSize": 16,
          "lineHeight": 24,
          "marginBottom": 32,
          "textAlign": "center",
        },
      ]
    }
  >
    Update now and keep earning points with your favorite brands!
  </Text>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityLabel="update-modal-update-btn"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="update-modal-update-btn"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#0066CC",
              "borderBottomLeftRadius": 30,
              "borderBottomRightRadius": 30,
              "borderTopLeftRadius": 30,
              "borderTopRightRadius": 30,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            },
            Object {
              "paddingBottom": 16,
              "paddingLeft": 32,
              "paddingRight": 32,
              "paddingTop": 16,
            },
          ]
        }
      >
        <View
          style={false}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "SFProDisplay-Heavy",
                    "fontSize": 16,
                  },
                  undefined,
                ],
              ]
            }
          >
            Update now
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="update-modal-close-btn"
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="update-modal-close-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "color": "#0066CC",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          I'll do this later
        </Text>
      </View>
    </View>
  </View>
</View>
`;
